### PR TITLE
[libcred] Add new port

### PR DESCRIPTION
**Pull request by Shadowrom2020**
- [x] Changes comply with the [maintainer guide](https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/contributing/maintainer-guide.md).
- [x] The name of the port matches an existing name for this component on https://repology.org/ if possible, and/or is strongly associated with that component on search engines.
- [x] Optional dependencies are resolved in exactly one way. For example, if the component is built with CMake, all `find_package` calls are REQUIRED, are satisfied by `vcpkg.json`'s declared dependencies, or disabled with [CMAKE_DISABLE_FIND_PACKAGE_Xxx](https://cmake.org/cmake/help/latest/variable/CMAKE_DISABLE_FIND_PACKAGE_PackageName.html).
- [x] The versioning scheme in `vcpkg.json` matches what upstream says.
- [x] The license declaration in `vcpkg.json` matches what upstream says.
- [x] The installed as the "copyright" file matches what upstream says.
- [x] The source code of the component installed comes from an authoritative source.
- [ ] The generated "usage text" is accurate. See [adding-usage](https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/examples/adding-usage.md) for context.
- [x] The version database is fixed by rerunning `./vcpkg x-add-version --all` and committing the result.
- [x] Only one version is in the new port's versions file.
- [x] Only one version is added to each modified port's versions file.

### CODE REVIEW

**Comment by dg0yt** [comment]
This is the 5th time you open a PR for this lib. You should always update the same PR (by updating the same GH branch without removing it first). 
Please mark it as draft as long as it is not ready for review.

**Comment by Shadowrom2020** [comment]
> This is the 5th time you open a PR for this lib. You should always update the same PR (by updating the same GH branch without removing it first). Please mark it as draft as long as it is not ready for review.

Sorry, won't happen again. the pr is marked as draft now,

File: ports/libcred/vcpkg.json [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+      "host": true,
+      "platform": "linux"
+    },
+    {
+      "name": "vcpkg-cmake",
+      "host": true
+    },
+    {
+      "name": "vcpkg-cmake-config",
+      "host": true
```
Comment (Cheney-W) [inline ports/libcred/vcpkg.json:20]:
```suggestion
```

**Review by Cheney-W — CHANGES_REQUESTED** [review]

File: ports/libcred/portfile.cmake [hunk 2]
```diff
@@ -0,0 +1,15 @@
+vcpkg_from_github(
+    OUT_SOURCE_PATH SOURCE_PATH
+    REPO mamba-org/libcred
+    REF ${VERSION}
+    SHA512 2b86cf0017c8e43f6d6eba378c00b56781b5d70b73d6251d8acde49f7fc698b41aaf2f0bb293211836ec9c3779828ae95cdc4041f3d0147c37352df007ad82bf
```
Comment (Cheney-W) [inline ports/libcred/portfile.cmake:5]:
```suggestion
    SHA512 77470b552cafc2506f9f1be56a2cd7aa412a4b568120037bf730273b5cc7c4dbabebb0abb6b192e3aef69912c6b5721d9e80b0cae0059f4fe814a5c0a8f3dcfb
```

**Comment by Cheney-W** [comment]
The command should be executed when `vcpkg x-add-version` is run for the second time:
`.\vcpkg x-add-version libcred --overwrite-version`

File: ports/libcred/vcpkg.json [hunk 3]
```diff
@@ -0,0 +1,15 @@
+{
+  "name": "libcred",
+  "version": "1.0.0",
+  "description": "a cross-platform credentials helper library",
+  "homepage": "https://github.com/mamba-org/libcred",
+  "license": "MIT",
+  "supports": "((windows & x64) | (windows & x86) | (linux & x86) | (linux & x64)) & !static & !uwp",
```
Comment (Cheney-W) [inline ports/libcred/vcpkg.json:7]:
```suggestion
  "supports": "!osx & !uwp & !arm & !(windows & staticcrt)",
```

File: ports/libcred/portfile.cmake [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+    HEAD_REF main
+)
+
+set(VCPKG_POLICY_DLLS_IN_STATIC_LIBRARY enabled)
+
+vcpkg_configure_meson(SOURCE_PATH "${SOURCE_PATH}")
+vcpkg_install_meson()
+vcpkg_install_copyright(FILE_LIST "${SOURCE_PATH}/LICENSE")
+vcpkg_fixup_pkgconfig()
+vcpkg_copy_pdbs()
```
Comment (Cheney-W) [inline ports/libcred/portfile.cmake:15]:
```suggestion
if (VCPKG_TARGET_IS_WINDOWS)
    vcpkg_check_linkage(ONLY_DYNAMIC_LIBRARY)
endif()

vcpkg_from_github(
    OUT_SOURCE_PATH SOURCE_PATH
    REPO mamba-org/libcred
    REF ${VERSION}
    SHA512 77470b552cafc2506f9f1be56a2cd7aa412a4b568120037bf730273b5cc7c4dbabebb0abb6b192e3aef69912c6b5721d9e80b0cae0059f4fe814a5c0a8f3dcfb
    HEAD_REF main
)

vcpkg_configure_meson(SOURCE_PATH "${SOURCE_PATH}")
vcpkg_install_meson()

vcpkg_install_copyright(FILE_LIST "${SOURCE_PATH}/LICENSE")

```

File: ports/libcred/vcpkg.json [hunk 5]
```diff
@@ -0,0 +1,15 @@
...
+  "description": "a cross-platform credentials helper library",
+  "homepage": "https://github.com/mamba-org/libcred",
+  "license": "MIT",
+  "supports": "!osx & !uwp & !arm & !(windows & staticcrt)",
+  "dependencies": [
+    {
+      "name": "libsecret",
+      "host": true,
+      "platform": "linux"
+    }
```
Comment (Cheney-W) [inline ports/libcred/vcpkg.json:13]:
```suggestion
    },
    {
      "name": "vcpkg-tool-meson",
      "host": true
    }
```

**Comment by Cheney-W** [comment]
Much thanks for your quick response.

**Review by Cheney-W — DISMISSED** [review]

**Comment by Shadowrom2020** [comment]
You're welcome. Sorry for the inconvenience, still I'm learning the quirks of vcpkg. Much has changed since my last contribution.

**Review by Cheney-W — DISMISSED** [review]

**Review by JavierMatosD — CHANGES_REQUESTED** [review]
This LGTM with the nitpicks. The building tests part is optional but you should definitely add `glib` as a dependency of the port. Mark "ready for review" when you've made the change.

File: ports/libcred/vcpkg.json [hunk 6]
```diff
@@ -0,0 +1,19 @@
...
+  "name": "libcred",
+  "version": "1.0.0",
+  "description": "a cross-platform credentials helper library",
+  "homepage": "https://github.com/mamba-org/libcred",
+  "license": "MIT",
+  "supports": "!osx & !uwp & !arm & !(windows & staticcrt) & !android",
+  "dependencies": [
+    {
+      "name": "libsecret",
+      "host": true,
```
Comment (dg0yt) [inline ports/libcred/vcpkg.json:11]: Also:
```suggestion
```

File: ports/libcred/vcpkg.json [hunk 7]
```diff
@@ -0,0 +1,19 @@
...
+      "platform": "linux"
```
[the remark this answers is not in the document]
Comment (Shadowrom2020) [reply ports/libcred/vcpkg.json:12]: libsecret depends on glib, there are no direct references to glib in libcred. As vcpkg installs libsecret and resolves glib automatically, I left thus out. Should I still add it to the dependencies?

File: ports/libcred/portfile.cmake [hunk 8]
```diff
@@ -0,0 +1,19 @@
+if (VCPKG_TARGET_IS_WINDOWS)
+    vcpkg_check_linkage(ONLY_DYNAMIC_LIBRARY)
+endif()
+
+vcpkg_from_github(
+    OUT_SOURCE_PATH SOURCE_PATH
+    REPO mamba-org/libcred
```
[the remark this answers is not in the document]
Comment (Shadowrom2020) [reply ports/libcred/portfile.cmake:7]: Missed that part, I'll add a patch for that later

**Review by Shadowrom2020 — COMMENTED** [review]
done

**Review by Cheney-W — APPROVED** [review]

**Comment by Shadowrom2020** [comment]
> This LGTM with the nitpicks. The building tests part is optional but you should definitely add `glib` as a dependency of the port. Mark "ready for review" when you've made the change.

Seems this is blocking the PR,

**Review by JavierMatosD — APPROVED** [review]

**Pull request merged**